libs/vkd3d/command.c

```diff
@@ -9154,86 +9174,96 @@ static void STDMETHODCALLTYPE d3d12_command_list_ClearUnorderedAccessViewUint(d3
     if (!vkd3d_clear_uav_info_from_desc(&args, &d))
         return;
 
-    if (args.has_view)
-        color = vkd3d_fixup_clear_uav_swizzle(list->device, d.view->info.view->format->dxgi_format, color);
+    if (d3d12_resource_is_texture(resource_impl) && !args.has_view)
+    {
+        /* Theoretically possibly for buggy application that tries to clear a buffer view with a texture resource.
+         * Safeguard against crash. */
+        ERR("Attempted to clear buffer with image resource.\n");
```

Comment:
`WARN`? That's what we usually do for potential game bugs to not spam messages by default.